app/models/user.rb

```diff
@@ -1322,7 +1323,7 @@ def active_roles(include_converted: true)
 
   def team_roles
     roles = []
-    self.current_team_members.each do |team_member|
+    self.official_current_team_members.each do |team_member|
```

Comment:
This is to avoid the intermediate stage where the WST admin exists in both groups and teams tables. During that time, the WST admin roles will get added in the normal roles way, and also in this method. It gets added in this method now and not earlier, because now the team is connected to a group (a team gets connected to a group if there is a group with same friendly id).

I don't want to get the roles added two times. So there are two solutions for this:
1. Consider only official_current_team_members here.
2. Remove teams & team_members related to WST admin in this same PR.

Since we were against 2 in previous PRs, I didn't go that way. But I assume it should be fine now because we have very less people in WST admin team and hence will be easy to revert if needed. What do you suggest? Continue with this way or go with 2?

Reply:
For a solution, I am inclined to go with 2. But then again, 2 is only a valid proposal _here_ because WST Admin is relatively small and relatively new -- that means smaller amount of `TeamMember` rows and much easier to debug. I wouldn't feel comfortable migrating the whole WST or WRT this way.

Reply:
So long story short, is there any clever approach of filtering out duplicate memberships?

Reply:
Okay, I made the user_group to return a team only if there are no actual roles. So, `current_team_members` should work now.

Reply:
Ignore my previous comment, instead I've added a method in user_group `roles_migrated?` which will check if there are any actual roles. If there are actual roles, it will assume that the roles are migrated, else it will look for team_members.